parse.go

```diff
@@ -74,6 +77,28 @@ func Parse(in io.Reader, bytesToRead int64, frameChan chan *frame.Frame) (Datase
 	return p.dataset, nil
 }
 
+// ParseWithOption parses the entire DICOM at the input io.Reader into a Dataset of DICOM Elements.
+// Allow user to use it with option.
+func ParseWithOption(in io.Reader, bytesToRead int64, frameChan chan *frame.Frame, opts ...ParseOption) (Dataset, error) {
```

Comment:
I think there is some overlap with #243 here. I think we can consider adding `opts ...ParseOption` to `Parse` directly